homework-submission/todo-app/src/todos/element.jsx

```diff
@@ -0,0 +1,120 @@
+import React, { Component } from 'react';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faCheck, faTrash, faPencilAlt, faTimes } from '@fortawesome/free-solid-svg-icons';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { fab } from '@fortawesome/free-brands-svg-icons'
+import {inject, observer} from 'mobx-react';
+
+
+
+ 
+library.add(fab, faCheck, faTrash, faPencilAlt, faTimes);
+
+@inject('TodosStore')
+@observer
+class Element extends Component {
+    state = {
+        id: this.props.id,
+        description: this.props.description,
+        deadline: this.props.deadline,
+        done: this.props.done,
+        editMode: false,
+        temporarySave: this.props.description
+     }
+
+     throwLineHandle = () => {
+        (!this.state.done) ? this.setState({done: true}) : this.setState({done: false});
+     }
+
+     turnUpdateMode = () => {
+        const temp = this.state.description;
+        this.setState({
+            editMode: !this.state.editMode,
+            temporarySave: temp
+        });
+     }
+
+     onChangeHandle = e => {
+         this.setState({[e.target.name]: e.target.value});
+     }
+
+     submitUpdate = e => {
+        e.preventDefault();
+        this.setState({editMode: !this.state.editMode});
+     }
+
+     handleCancel = () => {
+        this.setState({
+            description: this.state.temporarySave,
```

Comment:
It's a very nice Great name of `temporarySave`, But looking to the use of it, this is a bit different than the `handleCancel`, right?